internal/app/machined/pkg/controllers/network/platform_config.go

```diff
@@ -193,6 +198,23 @@ func (ctrl *PlatformConfigController) Run(ctx context.Context, r controller.Runt
 				return err
 			}
 		}
+
+		if err := safe.WriterModify(ctx, r, runtimeres.NewMetadataSpec(runtimeres.NamespaceName, runtimeres.PlatformMetadataID),
+			func(r *runtimeres.PlatformMetadata) error {
+				metaSpec := ctrl.V1alpha1Platform.Metadata(platformCtx, ctrl.PlatformState)
+
+				if metaSpec != nil {
+					*r.TypedSpec() = *metaSpec
+				} else {
+					*r.TypedSpec() = runtimeres.PlatformMetadataSpec{
+						Platform: ctrl.V1alpha1Platform.Name(),
+					}
+				}
```

Comment:
```suggestion
				*r.TypedSpec() = *ctrl.V1alpha1Platform.Metadata(platformCtx, ctrl.PlatformState)

				if r.TypedSpec() == nil {
					*r.TypedSpec() = runtimeres.PlatformMetadataSpec{
						Platform: ctrl.V1alpha1Platform.Name(),
					}
				}
```

this seems more clear

Reply:
Another issue here. What do we do if meta-server is not ready (network issue)? Or someone made a VM snapshot and create a new VM from it. 

I've moved the resource initialization to the "network ready" block. CCM won't change the node labels if the resource does not exist.

Reply:
Apply PlatformMetadata only when it's ready.